TauAnalysis/MCEmbeddingTools/plugins/EmbeddingLHEProducer.cc

```diff
@@ -424,6 +430,50 @@ void EmbeddingLHEProducer::rotate180(TLorentzVector &positiveLepton, TLorentzVec
   return;
 }
 
+void EmbeddingLHEProducer::InitialRecoCorrection(TLorentzVector &positiveLepton, TLorentzVector &negativeLepton) {
+  if (!InitialRecoCorrection_)
+    return;
+  edm::LogInfo("TauEmbedding") << "Applying initial reconstruction correction";
+  TLorentzVector Z = positiveLepton + negativeLepton;
+
+  edm::LogInfo("TauEmbedding") << "MuMinus before. Pt: " << negativeLepton.Pt() << " Mass: " << negativeLepton.M();
+  // std::cout << " MuMinus before. Pt: " << negativeLepton.Pt() << " Mass: " << negativeLepton.M() << " Energy: " << negativeLepton.E() << std::endl;
+  float diLeptonMass = (positiveLepton + negativeLepton).M();
+  if (diLeptonMass > 60. && diLeptonMass < 122.) {
+    // std::cout << "DiLeptonMass " << diLeptonMass << std::endl;
+    float zmass = 91.1876;
+    float correction_deviation =
+        5.;  // to ensure only a correction that drops corresponding to a Gaussian with mean zmass and std. dev. 5 GeV
+    double EmbeddingCorrection =
+        1.138;  // value derived by function fitting to fold embedded mass spectrum back to original selection when using mu -> mu embedding
+    EmbeddingCorrection =
+        EmbeddingCorrection /
+        (EmbeddingCorrection -
+         (EmbeddingCorrection - 1.) * exp(-pow((diLeptonMass - zmass), 2.) / (2. * pow(correction_deviation, 2.))));
+    EmbeddingCorrection = ((diLeptonMass + (EmbeddingCorrection - 1.) * zmass) / (diLeptonMass * EmbeddingCorrection));
+    double correctedpositiveLeptonEnergy = std::sqrt(
```

Comment:
I wonder if it's not better to factor the `EmbeddingCorrection` in front of the sqrt().  Then it will only show up in the denominator of the first term.

Reply:
Done in a8eae9f. This leads to minor precision changes in the values, which then lead to larger changes in the tau p_T after the simulation (for reconstructed taus up to 80GeV changes). However, since the changes come from numerical precision uncertainties and this is also a random process, this is fine.